platform/admin/core/admin-core-api/src/main/java/org/codice/ddf/ui/admin/api/impl/SystemPropertiesAdmin.java

```diff
@@ -0,0 +1,364 @@
+            updateProperty(SystemBaseUrl.HTTPS_PORT, updatedSystemProperties, systemDotProperties);
+            updateProperty(SystemInfo.ORGANIZATION, updatedSystemProperties, systemDotProperties);
+            updateProperty(SystemInfo.SITE_CONTACT, updatedSystemProperties, systemDotProperties);
+            updateProperty(SystemInfo.SITE_NAME, updatedSystemProperties, systemDotProperties);
+            updateProperty(SystemInfo.VERSION, updatedSystemProperties, systemDotProperties);
+
+            systemDotProperties.save();
+
+        } catch (IOException e) {
+            LOGGER.warn("Exception while writing to system.properties file.", e);
+        }
+
+        try {
+            Properties userDotProperties = new Properties(userPropertiesFile);
+
+            if (!userDotProperties.isEmpty()) {
+                String oldHostValue = userDotProperties.getProperty(oldHostName);
+
+                if (oldHostValue != null) {
+                    userDotProperties.remove(oldHostName);
+                    userDotProperties
+                            .setProperty(System.getProperty(SystemBaseUrl.HOST), oldHostValue);
+
+                    userDotProperties.save();
+                }
+
+            }
+
+        } catch (IOException e) {
+            LOGGER.warn("Exception while writing to users.properties file.", e);
+
+        }
+
+    }
+
+    private SystemPropertyDetails getSystemPropertyDetails(String key, String title,
+            String description, List<String> options) {
+        String defaultValue = "";
+        String value = "";
+        String property = System.getProperty(key);
+        if (property != null) {
+            defaultValue = property;
+            value = property;
+        }
+        return new SystemPropertyDetails(title, description, options, key, value, defaultValue);
+    }
+
+    private void updateProperty(String key, Map<String, String> updatedProperties,
+            Properties systemDotProperties) {
+        if (updatedProperties.containsKey(key)) {
+            String value = updatedProperties.get(key);
+            systemDotProperties.put(key, value);
+            System.setProperty(key, value);
+        }
+    }
+
+    private void configureMBean() {
+        mbeanServer = ManagementFactory.getPlatformMBeanServer();
+
+        try {
+            try {
+                mbeanServer.registerMBean(new StandardMBean(this, SystemPropertiesAdminMBean.class),
+                        objectName);
+            } catch (InstanceAlreadyExistsException e) {
+                mbeanServer.unregisterMBean(objectName);
+                mbeanServer.registerMBean(new StandardMBean(this, SystemPropertiesAdminMBean.class),
+                        objectName);
+            }
+        } catch (Exception e) {
+            // something here
+        }
+    }
+
+    @Override
+    public List<String> setSystemCerts(Map<String, Object> certProps) {
```

Comment:
It seems like we could just hit the jolokia KeystoreEditor endpoints directly instead of creating your own custom mbean that just calls the KeystoreEditor stuff directly. Then there is no hard dependency between these two things.